sklearn/utils/__init__.py

```diff
@@ -288,7 +288,7 @@ def safe_sqr(X, copy=True):
         X.data **= 2
     else:
         if copy:
-            X = X ** 2
+            X **= 2
```

Comment:
This is actually really really bad. Look at the logic before and after. Now the if is meaningless.